common/store.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jinzhu/gorm"
+	_ "github.com/mattn/go-sqlite3"
+	"log"
+	"os"
+	"strconv"
+)
+
+// MultiError adapts GORM (ORM - see https://github.com/jinzhu/gorm) array of errors found in GetErrors()
+// to a single error interface.
+// GORMdoes not return errors at every turn. It accumulates them and returns
+// them whenever you feel like calling GetErrors() (https://godoc.org/github.com/jinzhu/gorm#DB.GetErrors).
+// Since this is not consistent  with the rest of the code, I prefer to isolate it
+// here and make an adapter.
+type MultiError struct {
+	errors []error
+}
+
+func MakeMultiError(errors []error) error {
+	if errors == nil {
+
+		return nil
+	}
+	if len(errors) == 0 {
+
+		return nil
+	}
+
+	return &MultiError{errors}
+}
+
+func (m *MultiError) Error() string {
+	s := ""
+	for i := range m.errors {
+		if len(s) > 0 {
+			s += "; "
+		}
+		s += m.errors[i].Error()
+	}
+	return s
+}
+
+// Stores information needed for a DB connection.
+type StoreConfig struct {
+	Host     string
+	Port     uint64
+	Username string
+	Password string
+	Database string
+	Type     string
+}
+
+// MakeStoreConfig creates StoreConfig object from a map.
+func makeStoreConfig(configMap map[string]interface{}) StoreConfig {
+	storeConfig := StoreConfig{}
+	storeConfig.Type = configMap["type"].(string)
+	if configMap["host"] != nil {
+		storeConfig.Host = configMap["host"].(string)
+	}
+	if configMap["port"] != nil {
+		portStr := configMap["port"].(string)
+		port, err := strconv.ParseUint(portStr, 10, 64)
+		if err != nil {
+			log.Printf("Error parsing %s", portStr)
+		} else {
+			storeConfig.Port = port
+		}
+	}
+	if configMap["username"] != nil {
+		storeConfig.Username = configMap["username"].(string)
+	}
+	if configMap["password"] != nil {
+		storeConfig.Password = configMap["password"].(string)
+	}
+	storeConfig.Database = configMap["database"].(string)
+	return storeConfig
+}
+
+// Defines generic store interface that can be used
+// by any service for persistence.
+type Store interface {
```

Comment:
Where is this type used, if at all? Searching for 'Store' on this page yields more than 600 results, so a little hint might be useful...